src/performance/metrics/PerformanceUtil.cpp

```diff
@@ -0,0 +1,1369 @@
+                                 " where is_host_reporter='true'";
+    std::vector<vector<pair<string, string>>> placeList = perfDb.runSelect(placeLoadQuery);
+    std::vector<vector<pair<string, string>>>::iterator placeListIterator;
+
+    for (placeListIterator = placeList.begin(); placeListIterator != placeList.end(); ++placeListIterator) {
+        vector<pair<string, string>> place = *placeListIterator;
+        Place placeInfo;
+        std::string host;
+        std::string requestResourceAllocation = "false";
+
+        placeInfo.ip = place.at(0).second;
+        placeInfo.user = place.at(1).second;
+        placeInfo.serverPort = place.at(2).second;
+        placeInfo.isMaster = place.at(3).second;
+        placeInfo.isHostReporter = place.at(4).second;
+        placeInfo.hostId = place.at(5).second;
+        placeInfo.placeId = place.at(6).second;
+
+        hostReporterList.push_back(placeInfo);
+    }
+
+    return hostReporterList;
+
+}
+
+int PerformanceUtil::collectSLAResourceConsumption(std::vector<Place> placeList, std::string graphId,
+        std::string masterIP, int elapsedTime) {
+    /*string slaCategoryId;
+    std::vector<std::future<long>> intermRes;
+    PlacesToNodeMapper placesToNodeMapper;
+
+    std::string placeLoadQuery = "select ip, user, server_port, is_master, is_host_reporter,host_idhost,idplace from place"
+                                 " where is_host_reporter='true'";
+    std::vector<vector<pair<string, string>>> placeList = perfDb.runSelect(placeLoadQuery);
+    std::vector<vector<pair<string, string>>>::iterator placeListIterator;
+
+    std::string categoryQuery = "SELECT id from sla_category where command='" + command + "' and category='" + category + "'";
+
+    std::vector<vector<pair<string, string>>> categoryResults = perfDb.runSelect(categoryQuery);
+
+    if (categoryResults.size() == 1) {
+        slaCategoryId = categoryResults[0][0].second;
+    } else {
+        scheduler_logger.log("Invalid SLA " + category + " for " + command + " command", "error");
+        return 0;
+    }*/
+
+    std::vector<Place>::iterator placeListIterator;
+
+    for (placeListIterator = placeList.begin(); placeListIterator != placeList.end(); ++placeListIterator) {
+        Place place = *placeListIterator;
+        std::string host;
+        std::string requestResourceAllocation = "false";
+
+        std::string ip = place.ip;
+        std::string user = place.user;
+        std::string serverPort = place.serverPort;
+        std::string isMaster = place.isMaster;
+        std::string isHostReporter = place.isHostReporter;
+        std::string hostId = place.hostId;
+        std::string placeId = place.placeId;
+
+        if (ip.find("localhost") != std::string::npos || ip.compare(masterIP) == 0) {
+            host = ip;
+        } else {
+            host = user + "@" + ip;
+        }
+
+        /*if (isHostReporter.find("true") != std::string::npos) {
```

Comment:
Why do we have the commented lines here?

Reply:
Removed the commented section from the code

Reply:
Ok